test/libsolidity/syntaxTests/constants/pure_non_rational.sol

```diff
@@ -0,0 +1,11 @@
+// Tests that the ConstantEvaluator does not crash for pure non-rational functions.
+// Currently it does not evaluate such functions, but this may change in the future
+// causing a division by zero error for a.
+contract C {
+	uint constant a = 1 / (uint(keccak256([0])[0]) - uint(keccak256([0])[0]));
```

Comment:
Please change this to
```
	uint constant a = 1 / (uint(keccak256([0])[0]) - uint(keccak256([0])[0]));
	uint constant b = 1 / uint(keccak256([0]));
	uint constant c = uint(keccak256([0]));
	uint[c] mem;
```
so that all constant variables themselves are at least type-correct.

Reply:
(i.e. add explicit conversions to `uint`).

Reply:
Done.